bgpd/bgp_fsm.c

```diff
@@ -1133,6 +1184,38 @@ int bgp_stop(struct peer *peer)
 					peer->nsf[afi][safi] = 0;
 		}
 
+		/* If peer reset before receiving EOR, decrement EOR count and
+		 * cancel the selection deferral timer if there are no
+		 * pending EOR messages to be received
+		 */
+		if (BGP_PEER_GRACEFUL_RESTART_CAPABLE(peer)) {
+			FOREACH_AFI_SAFI (afi, safi) {
+				if (peer->afc_nego[afi][safi] &&
+				    !CHECK_FLAG(peer->af_sflags[afi][safi],
+						PEER_STATUS_EOR_RECEIVED)) {
+					gr_info = &bgp->gr_info[afi][safi];
+
+					if (gr_info && (gr_info->eor_required))
+						gr_info->eor_required--;
+
+					if (gr_info && BGP_DEBUG(update,
+						UPDATE_OUT))
+						zlog_debug(
+							"peer %s, EOR_required %d",
+							peer->host,
```

Comment:
can this be in format like many others? `"%s: ...", peer->host`

Reply:
"peer <name of the peer>" is more readable than "<name of the peer> peer"  ?